packages/server/src/plugin/subscriptionCallback/index.ts

```diff
@@ -0,0 +1,655 @@
+import type { Logger } from '@apollo/utils.logger';
+import { GraphQLError, subscribe, type ExecutionResult } from 'graphql';
+import fetch, { type Response } from 'node-fetch';
+import { ensureGraphQLError } from '../../errorNormalize.js';
+import type { ApolloServerPlugin } from '../../externalTypes/index.js';
+import { internalPlugin } from '../../internalPlugin.js';
+import { HeaderMap } from '../../utils/HeaderMap.js';
+
+export interface ApolloServerPluginSubscriptionCallbackOptions {
+  heartbeatIntervalMs?: number;
+  logger?: Logger;
+}
+
+export function ApolloServerPluginSubscriptionCallback(
+  options: ApolloServerPluginSubscriptionCallbackOptions = Object.create(null),
+): ApolloServerPlugin {
+  const subscriptionManager = new SubscriptionManager(options);
+  const logger = options.logger
+    ? prefixedLogger(options.logger, 'SubscriptionCallback')
+    : undefined;
+
+  return internalPlugin({
+    __internal_plugin_id__: 'SubscriptionCallback',
+    __is_disabled_plugin__: false,
+    async requestDidStart({ request }) {
+      // If it's not a callback subscription, ignore the request.
+      if (!request?.extensions?.subscription) return;
```

Comment:
can save this value to a const to avoid the `!`s later.

Reply:
[ee1c3ab](https://github.com/apollographql/apollo-server/pull/7617/commits/ee1c3ab4a058ce2a0bbfe0d1bdaf33cd49387beb)